src/tools/fuzzing/fuzzing.cpp

```diff
@@ -3331,7 +3327,25 @@ HeapType TranslateToFuzzReader::getSubType(HeapType type) {
         break;
     }
   }
-  // TODO: nontrivial types as well.
+  // Look for an interesting subtype. First, compute the possible interesting
+  // heap types, if we haven't already.
+  auto iter = interestingHeapSubTypes.find(type);
+  if (iter == interestingHeapSubTypes.end()) {
+    std::vector<HeapType> subTypes;
+    for (auto possible : interestingHeapTypes) {
+      // We avoid storing the type itself among its subtypes since there is
+      // already a good chance to return the type itself from this function.
+      if (possible != type && HeapType::isSubType(possible, type)) {
+        subTypes.push_back(possible);
+      }
+    }
```

Comment:
This loop is `O(n^2)` where `n` is the number of heap types because `HeapType::isSubType` is linear in the number of heap types (in the worst case).

In contrast, walking the type graph once and collecting all the subtypes for all types naively without using `HeapType::isSubType` would be `O(n^2)`, but would collect the subtypes for _all_ types, not just a single type.

(It's also possible to collect all the subtypes for all types in linear time using a fancy flattened representation of the subtype tree, but the complexity is certainly not worth it.)

Reply:
Now I see, thanks! I had assumed not many would be called, and didn't think it through that even a single iteration is quadratic.

Rewritten to be efficient.